test/tests.test.js

```diff
@@ -15,173 +13,149 @@ const urlQuery = 'a=b&c=d'
 const httpScheme = 'http'
 const httpsScheme = 'https'
 
-test('parses a full URI', (t) => {
-  t.plan(10)
-  let port
+test('parses a full URI', async (t) => {
+  t.plan(11)
   const fastify = Fastify()
 
   fastify
     .register(plugin)
     .after((err) => {
-      if (err) t.error(err)
+      t.assert.ok(!err)
```

Comment:
We should update this to [`ifError`](https://nodejs.org/api/assert.html#assertiferrorvalue) we use this when testing the error argument in callbacks in fastify repo
```suggestion
      t.assert.ifError(err)
```

Reply:
I tend to disagree on this. If the signature of the `after` function changes, returning something else than an error, with `ok` the test will fail while with `ifError` will always pass.
But ok, I'll change it.